internal/service-scraper/scraper.go

```diff
@@ -79,9 +83,41 @@ func (s *Scraper) GetInstances(ctx context.Context) ([]ScraperDiscoveredTarget,
 				},
 			}
 
-			awsRdsInstances = append(awsRdsInstances, instanceTarget)
+			instances = append(instances, instanceTarget)
 		}
 	}
 
-	return awsRdsInstances, nil
+	return instances, nil
+}
+
+func (s *Scraper) getClusterAwsClient(ctx context.Context, target ScrapingTarget) (*rds.Client, error) {
+	conf, err := awsConfig.LoadDefaultConfig(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// basic AWS client without assuming role
+	if target.ClusterAssumeRoleArn == nil {
+		return rds.NewFromConfig(conf), nil
+	}
+
+	targetArn, err := arn.Parse(target.ClusterArn)
+	if err != nil {
+		return nil, fmt.Errorf("invalid RDS cluster ARN: %s", target.ClusterArn)
+	}
+
+	// create a new AWS client with assumed role
+	stsClient := sts.NewFromConfig(conf)
+	assumedRoleCred := stscreds.NewAssumeRoleProvider(stsClient, *target.ClusterAssumeRoleArn)
+
+	assumedRoleConfig := conf
+	assumedRoleConfig.Credentials = aws.NewCredentialsCache(assumedRoleCred)
+
+	// this must be here because in AWS RDS API there is error,
```

Comment:
RDS API is regional. 
eu-central-1 has its own domain and doesn't know anything about cluster in other regions, the client has to be always regional for RDS

Reply:
Yea you are right!